superset-frontend/src/components/ListView/utils.ts

```diff
@@ -189,27 +221,48 @@ export function useListViewState({
   );
 
   const [internalFilters, setInternalFilters] = useState<InternalFilter[]>(
-    query.filters || [],
+    query.filters && initialFilters.length
+      ? mergeCreateFilterValues(initialFilters, query.filters)
+      : [],
   );
 
   useEffect(() => {
     if (initialFilters.length) {
       setInternalFilters(
-        mergeCreateFilterValues(initialFilters, query.filters || []),
+        mergeCreateFilterValues(
+          initialFilters,
+          query.filters ? query.filters : {},
+        ),
       );
     }
   }, [initialFilters]);
 
   useEffect(() => {
+    // From internalFilters, produce a simplified obj
+    const filterObj = {};
+
+    internalFilters.forEach(filter => {
+      if (
+        filter.value !== undefined &&
+        (typeof filter.value !== 'string' || filter.value.length > 0)
+      ) {
+        filterObj[filter.id] = [filter.value, filter.operator];
```

Comment:
As per your TODO above, we could probably even get rid of the operator since that value is static and set in code. I doubt we want to support swapping filter operators in the url.